modules/plugins/comments/comment-nvim/config.nix

```diff
@@ -14,35 +13,30 @@ in {
   config = mkIf cfg.enable {
     vim.startPlugins = ["comment-nvim"];
 
-    vim.maps.normal = mkMerge [
-      (mkBinding cfg.mappings.toggleOpLeaderLine "<Plug>(comment_toggle_linewise)" mappings.toggleOpLeaderLine.description)
-      (mkBinding cfg.mappings.toggleOpLeaderBlock "<Plug>(comment_toggle_blockwise)" mappings.toggleOpLeaderBlock.description)
-
-      (mkExprBinding cfg.mappings.toggleCurrentLine ''
-          function()
-            return vim.api.nvim_get_vvar('count') == 0 and '<Plug>(comment_toggle_linewise_current)'
-                    or '<Plug>(comment_toggle_linewise_count)'
-          end
-        ''
-        mappings.toggleCurrentLine.description)
-      (mkExprBinding cfg.mappings.toggleCurrentBlock ''
-          function()
-            return vim.api.nvim_get_vvar('count') == 0 and '<Plug>(comment_toggle_blockwise_current)'
-                    or '<Plug>(comment_toggle_blockwise_count)'
-          end
-        ''
-        mappings.toggleCurrentBlock.description)
-    ];
-
-    vim.maps.visual = mkMerge [
-      (mkBinding cfg.mappings.toggleSelectedLine "<Plug>(comment_toggle_linewise_visual)" mappings.toggleSelectedLine.description)
-      (mkBinding cfg.mappings.toggleSelectedBlock "<Plug>(comment_toggle_blockwise_visual)" mappings.toggleSelectedBlock.description)
-    ];
-
-    vim.pluginRC.comment-nvim = entryAnywhere ''
-      require('Comment').setup({
-        mappings = { basic = false, extra = false, },
-      })
-    '';
+    vim.lazy.plugins.comment-nvim = {
+      package = "comment-nvim";
+      setupModule = "Comment";
+      inherit (cfg) setupOpts;
+      keys = [
+        (mkLznBinding ["n"] cfg.mappings.toggleOpLeaderLine "<Plug>(comment_toggle_linewise)" mappings.toggleOpLeaderLine.description)
+        (mkLznBinding ["n"] cfg.mappings.toggleOpLeaderBlock "<Plug>(comment_toggle_blockwise)" mappings.toggleOpLeaderBlock.description)
+        (mkLznExprBinding ["n"] cfg.mappings.toggleCurrentLine ''
+            function()
+              return vim.api.nvim_get_vvar('count') == 0 and '<Plug>(comment_toggle_linewise_current)'
+                      or '<Plug>(comment_toggle_linewise_count)'
+            end
+          ''
+          mappings.toggleCurrentLine.description)
+        (mkLznExprBinding ["n"] cfg.mappings.toggleCurrentBlock ''
+            function()
+              return vim.api.nvim_get_vvar('count') == 0 and '<Plug>(comment_toggle_blockwise_current)'
+                      or '<Plug>(comment_toggle_blockwise_count)'
+            end
+          ''
+          mappings.toggleCurrentBlock.description)
+        (mkLznBinding ["x"] cfg.mappings.toggleSelectedLine "<Plug>(comment_toggle_linewise_visual)" mappings.toggleSelectedLine.description)
+        (mkLznBinding ["x"] cfg.mappings.toggleSelectedBlock "<Plug>(comment_toggle_blockwise_visual)" mappings.toggleSelectedBlock.description)
```

Comment:
I'm pretty sure
```suggestion
        (mkLznBinding ["v"] cfg.mappings.toggleSelectedLine "<Plug>(comment_toggle_linewise_visual)" mappings.toggleSelectedLine.description)
        (mkLznBinding ["v"] cfg.mappings.toggleSelectedBlock "<Plug>(comment_toggle_blockwise_visual)" mappings.toggleSelectedBlock.description)
```

Reply:
the default keybind is `gc`, which would interfere with typing if we also bind it to select mode...